File: src/main/Java/boj_step20/boj26069.java
import java.io.*;
import java.util.HashSet;
import java.util.StringTokenizer;

public class boj26069 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(reader.readLine());
        HashSet<String> list = new HashSet<>();
        int count = 0;
        while (n-- > 0) {
            StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
            String a = tokenizer.nextToken();
            String b = tokenizer.nextToken();
            if ((a.equals("ChongChong") || b.equals("ChongChong")) && !list.contains("ChongChong")) {
                list.add(a);
                list.add(b);
                count += list.size();
            }
            if (list.contains(a) && !list.contains(b)) { list.add(b); count++; }
            else if (list.contains(b) && !list.contains(a)) { list.add(a); count++; }
        }
        writer.append(String.valueOf(count)).close();
    }
}

File: src/main/boj_step20/boj26069.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine().toInt()
    val peoples = HashSet<String>()
    var count = 0
    repeat(n) {
        val (a, b) = r.readLine().split(" ").map { it }
        if ((a == "ChongChong" || b == "ChongChong") && "ChongChong" !in peoples) {
            peoples.add(a)
            peoples.add(b)
            count += peoples.size
        }
        if (a in peoples && b !in peoples) {
            peoples.add(b)
            count++
        } else if (a !in peoples && b in peoples) {
            peoples.add(a)
            count++
        }
    }
    w.write(count.toString())
    w.close()
}